src/WebAppSSLManager/CertificatesHelper.cs

```diff
@@ -125,6 +125,9 @@ public static async Task<bool> GetCertificateAsync()
             await AzureHelper.RemoveDNSVerificationTXTRecord(recordName); //to be sure we start clean
             await AzureHelper.CreateDNSVerificationTXTRecord(recordName, dnsTxt);
 
+            _logger.LogInformation($"   Waiting before validating DNS authorization challenge...");
+            Thread.Sleep(Settings.WaitTimeBeforeValidate);
```

Comment:
```suggestion
            _logger.LogInformation($"   Waiting {Settings.WaitTimeBeforeValidate.TotalSeconds} seconds before validating DNS authorization challenge...");
            await Task.Delay(Settings.WaitTimeBeforeValidate.TotalMilliseconds);
```

I would rather use a non-blocking option like this.
Also, for this approach, probably better changing from `TimeSpan` to `int` and have it in seconds